core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/variable/listener/support/Notifiable.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.variable.listener.support;
+
+public interface Notifiable {
```

Comment:
I'd say it's for the same reason why the `AbstractVariableListener` cares about `resetWorkingSolution()`.

I tried to come up with a better name but I couldn't. Instead of that, I added some Javadoc to explain the relationships between listeners, notifiables, notifications, and their specializations.